api/src/lib/expand-function/imports/context-for-import.ts

```diff
@@ -0,0 +1,193 @@
+    logger.warn(
+      `[followImport] Resolved import path does not exist: ${resolvedPath}`,
+    );
+    return null;
+  }
+
+  await openFile(tsserver, resolvedPath);
+
+  const importedFileContent = await fs.promises.readFile(resolvedPath, "utf-8");
+  const importedSourceFile = ts.createSourceFile(
+    resolvedPath,
+    importedFileContent,
+    ts.ScriptTarget.Latest,
+    true,
+  );
+
+  // Find the definition in the imported file
+  const importClause = importNode.importClause;
+  if (importClause) {
+    let identifierToFind: string | undefined;
+
+    if (
+      importClause.name &&
+      importClause.name.text === identifierNode.getText()
+    ) {
+      // Default import
+      identifierToFind = importClause.name.text;
+    } else if (
+      importClause.namedBindings &&
+      ts.isNamedImports(importClause.namedBindings)
+    ) {
+      // Named import
+      const namedImport = importClause.namedBindings.elements.find(
+        (element) => element.name.text === identifierNode.getText(),
+      );
+      if (namedImport) {
+        identifierToFind =
+          namedImport.propertyName?.text || namedImport.name.text;
+      }
+    }
+
+    if (identifierToFind) {
+      logger.debug(
+        `[debug][followImport] Identifier to find in file we're importing from: ${identifierToFind}`,
+      );
+      const importedNode = findExportedDeclaration(
+        importedSourceFile,
+        identifierToFind,
+      );
+      if (importedNode) {
+        const result = getDefinitionText(importedNode, importedSourceFile);
+
+        const definitionText = result?.text;
+        const definitionType = result?.type;
+
+        return {
+          uri: getFileUri(resolvedPath),
+          range: {
+            start: importedSourceFile.getLineAndCharacterOfPosition(
+              importedNode.getStart(),
+            ),
+            end: importedSourceFile.getLineAndCharacterOfPosition(
+              importedNode.getEnd(),
+            ),
+          },
+          text: definitionText,
+          type: definitionType,
+        };
+      }
+    }
+  }
+
+  return null;
+}
+
+function findExportedDeclaration(
+  sourceFile: ts.SourceFile,
+  identifierName: string,
+): ts.Node | undefined {
+  return sourceFile.statements.find((statement) => {
+    if (
+      ts.isExportAssignment(statement) &&
+      ts.isIdentifier(statement.expression)
+    ) {
+      return statement.expression.text === identifierName;
+    }
+    if (
+      ts.isFunctionDeclaration(statement) ||
+      ts.isVariableStatement(statement)
+    ) {
+      const modifiers = ts.getModifiers(statement);
+      return (
+        modifiers?.some(
+          (modifier) => modifier.kind === ts.SyntaxKind.ExportKeyword,
+        ) &&
+        (ts.isFunctionDeclaration(statement)
+          ? statement.name?.text === identifierName
+          : statement.declarationList.declarations[0].name.getText() ===
+            identifierName)
+      );
+    }
```

Comment:
This is quite dense code to read, how about splitting it up a bit? Something like:

```suggestion
      const hasExportModifier = modifiers?.some(
          (modifier) => modifier.kind === ts.SyntaxKind.ExportKeyword,
        );

      const matchesIdentifier =  (ts.isFunctionDeclaration(statement)
          ? statement.name?.text === identifierName
          : statement.declarationList.declarations[0].name.getText() ===
            identifierName);
            
      return hasExportModifier && matchesIdentifier;
    }
```